### PR TITLE
Create "clean_residual_files" target in Makefile

### PR DESCRIPTION
**Pull request by eksperimental**
This commits creates new clean targets.
It removes all the files in the .gitignore file.
It also deals with situations where `make test` was interrumpted, leaving files that are neither gonna be ignored by .gitignore nor gonna be removed by `make clean`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /doc
 /ebin
-/lib/*/ebin/*
-/lib/*/tmp
+/lib/**/ebin
```
Comment (josevalim) [inline .gitignore]: I don't think those new entries are better.

**Comment by josevalim** [comment]
Hi @eksperimental. I believe this commit adds a lot of indirection. I would prefer to explicitly list entries and rely on commands that use find and so on. Thank you.

File: .gitignore (hunk 1 above)
Comment (eksperimental) [inline .gitignore]: why you think they are not.
It solves the problem of interrupted tests and residual folders created in fixtures, that are deeper than one dir under lib/

**Comment by eksperimental** [comment]
Have a look at it again @josevalim 
I have removed all the indirection except for the one target I created `clean_residual_files`.
It is the alternative to `make clean` when you want to clean up the directory after a crash or an interrupted test, but you don't want to remove the beam files and recompile again.

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,8 @@
 /doc
 /ebin
-/lib/*/ebin/*
-/lib/*/tmp
+/lib/**/ebin/
```
Comment (josevalim) [inline .gitignore]: Why this change still? Thank you.

File: Makefile [hunk 3]
```diff
@@ -114,19 +114,25 @@ install: compile
...
 clean:
 	cd lib/elixir && $(REBAR) clean
-	rm -rf ebin
-	rm -rf lib/*/ebin
-	rm -rf lib/elixir/test/ebin
-	rm -rf lib/*/tmp
-	rm -rf lib/mix/test/fixtures/git_repo
-	rm -rf lib/mix/test/fixtures/deps_on_git_repo
-	rm -rf lib/mix/test/fixtures/git_rebar
+	find ./lib/ -type d -name 'ebin' -exec rm -rf {} +
```
Comment (josevalim) [inline Makefile]: Please remove the `find`.

File: Makefile [hunk 4]
```diff
@@ -114,19 +114,25 @@ install: compile
...
 clean_exbeam:
 	$(Q) rm -f lib/*/ebin/Elixir.*.beam
 
+clean_residual_files:
+	rm -rf lib/mix/test/fixtures/deps_on_git_repo/
+	rm -rf lib/mix/test/fixtures/git_rebar/
+	rm -rf lib/mix/test/fixtures/git_repo/
+	rm -rf lib/mix/test/fixtures/git_sparse_repo/
+	rm -rf lib/*/tmp/
+	find ./lib/ -type d -name '_build' -exec rm -rf {} +
```
Comment (josevalim) [inline Makefile]: Please remove `find`.

File: Makefile [hunk 5]
```diff
@@ -114,19 +114,25 @@ install: compile
...
+	find . -type f -name 'erl_crash.dump' -exec rm {} +
```
Comment (josevalim) [inline Makefile]: Please remove the `find`. No need to clean up the crash dumps for now.

**Comment by eksperimental** [comment]
@josevalim please give a new review to everything

File: Makefile [hunk 6]
```diff
@@ -116,17 +116,25 @@ clean:
...
 
+clean_residual_files:
+	rm -rf lib/*/_build/
+	rm -rf lib/*/tmp/
+	rm -rf lib/elixir/test/ebin/
+	rm -rf lib/mix/test/fixtures/deps_on_git_repo/
+	rm -rf lib/mix/test/fixtures/git_rebar/
+	rm -rf lib/mix/test/fixtures/git_repo/
+	rm -rf lib/mix/test/fixtures/git_sparse_repo/
+	rm -f erl_crash.dump
```
Comment (eksperimental) [inline Makefile]: @josevalim not sure if you didn't want to remote erl_crash.dump in subfolder, or that you didn't want to remove erl_crash.dump at all

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
